src/sql/sql-unary/sql-unary.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SqlBase, SqlBaseValue } from '../index';
+
+export interface UnaryExpressionValue extends SqlBaseValue {
+  keyword?: string;
+  argument?: SqlBase;
+  expressionType?: string;
+}
+export class SqlUnary extends SqlBase {
+  static wrapInQuotes(thing: string, quote: string): string {
+    return `${quote}${thing}${quote}`;
+  }
+
+  public expressionType?: string;
+  public keyword?: string;
+  public argument?: SqlBase;
+
+  constructor(options: UnaryExpressionValue) {
+    super(options, 'unaryExpression');
+    this.expressionType = options.expressionType;
+    this.keyword = options.keyword;
+    this.argument = options.argument;
+  }
+
+  public valueOf() {
+    const value: UnaryExpressionValue = super.valueOf();
+    value.expressionType = this.expressionType;
+    value.keyword = this.keyword;
+    value.argument = this.argument;
+    return value;
+  }
+
+  public toRawString(): string {
+    if (!this.argument) {
+      throw new Error('Could not make raw string');
+    }
+    return this.keyword + this.innerSpacing.postKeyword + this.argument.toString();
+  }
+}
+
+SqlBase.register('unaryExpression', SqlUnary);
```

Comment:
This may be a little beyond the scope of this specific pull request but this `SqlBase.register` stuff is _slightly_ dangerous. You could make it safer by (in the case of this file) defining `'unaryExpression'` as a constant and plugging that into both `super` and `register`, and it might be nice to give `register` a mechanism for preventing type name collisions. (I'd really love to do away with the whole thing and get the constructor polymorphically but I don't know if that's very easy to do here.)